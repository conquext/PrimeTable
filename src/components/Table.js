import React from "react";
import Row from "./Row";

const Table = ({ nPrimes }) => {
  let rowMul;

  return (
    <div className="prime-table">
      <Row rList={["", ...nPrimes]} className="odd-row border-bottom" />
      <div>
        {nPrimes.map((list, index) => {
          rowMul = nPrimes.map(np => np * list);
          return (
            <Row
              key={index}
              rList={[list, ...rowMul]}
              className={index % 2 !== 0 ? "odd-row" : "even-row"}
            />
          );
        })}
      </div>
    </div>
  );
};

export default Table;
