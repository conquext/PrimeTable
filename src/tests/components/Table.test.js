import React from "React";
import ReactShallowRenderer from "react-test-renderer/shallow";

import Table from "../../components/Table";

const nPrimes = [2, 3, 5, 7, 11];

test("should render Table correctly", () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<Table nPrimes={nPrimes} />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
