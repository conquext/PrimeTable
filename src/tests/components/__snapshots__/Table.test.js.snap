// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Table correctly 1`] = `
<div
  className="prime-table"
>
  <Row
    className="odd-row border-bottom"
    rList={
      Array [
        "",
        2,
        3,
        5,
        7,
        11,
      ]
    }
  />
  <div>
    <Row
      className="even-row"
      rList={
        Array [
          2,
          4,
          6,
          10,
          14,
          22,
        ]
      }
    />
    <Row
      className="odd-row"
      rList={
        Array [
          3,
          6,
          9,
          15,
          21,
          33,
        ]
      }
    />
    <Row
      className="even-row"
      rList={
        Array [
          5,
          10,
          15,
          25,
          35,
          55,
        ]
      }
    />
    <Row
      className="odd-row"
      rList={
        Array [
          7,
          14,
          21,
          35,
          49,
          77,
        ]
      }
    />
    <Row
      className="even-row"
      rList={
        Array [
          11,
          22,
          33,
          55,
          77,
          121,
        ]
      }
    />
  </div>
</div>
`;
